### PR TITLE
UPBGE: Import Benoit Bolsee VideoTexture commit into upbge

### PR DESCRIPTION
**Pull request by youle31**
notes: in RAS_OpenglRasterizer.cpp I included KX_Globals to get canvas
(KX_GetActiveEngine->GetCanvas()) for
RAS_OpenGLRasterizer::CreateOffScreen function.

I noticed a performances decrease in water demo file.

I made no test of the new features for the moment.

### CODE REVIEW

File: intern/moto/include/MT_Matrix4x4.h [hunk 1]
```diff
@@ -131,6 +131,16 @@ class MT_Matrix4x4 {
     }
 
 	/**
+	 * Scale the rows of this matrix with x, y, z, w respectively.
```
Comment (panzergame) [inline intern/moto/include/MT_Matrix4x4.h]: To reduce cleanup work you can commit this modification in a independant commit

**Comment by youle31** [comment]
As I have different performance results than the author of the patch, could someone test this files with this branch and the master to see if there is a performance difference (water with fbo works only with this branch). Also, if you want to make a complete test, you can compare the framerate with the one in BF Blender 2.77a.

If you can post your results it would be very nice. Thanks!

Files: http://www.pasteall.org/blend/42429 (without fbo)
http://www.pasteall.org/blend/42430 (with fbo)

**Comment by panzergame** [comment]
Sorry for delay:

Without FBO: 6.7ms per frame
With FBO: 6.7ms per frame

**Comment by youle31** [comment]
and in BF blender 2.77a without fbo?

**Comment by panzergame** [comment]
The benoit's commits was merged, it request now a big cleanup.

**Pull request closed without merging**